modules/swagger-codegen/src/main/resources/rust/api.mustache

```diff
@@ -1,24 +1,17 @@
 {{>partial_header}}
-use std::rc::Rc;
-use std::borrow::Borrow;
-use std::borrow::Cow;
-use std::collections::HashMap;
-
 use hyper;
+use hyper::header::{HeaderName, HeaderValue};
+use hyper::rt::{Future, Stream};
 use serde_json;
-use futures;
-use futures::{Future, Stream};
-
-use hyper::header::UserAgent;
 
-use super::{Error, configuration};
+use super::{Error, configuration, client};
 
-pub struct {{{classname}}}Client<C: hyper::client::Connect> {
-    configuration: Rc<configuration::Configuration<C>>,
+pub struct {{{classname}}}Client<'a, C: hyper::client::connect::Connect> {
+    configuration: &'a configuration::Configuration<C>,
 }
 
-impl<C: hyper::client::Connect> {{{classname}}}Client<C> {
-    pub fn new(configuration: Rc<configuration::Configuration<C>>) -> {{{classname}}}Client<C> {
```

Comment:
I don't understand why `Rc` was being used. I updated it to be a plain reference.

Reply:
The `Rc` was there to omit the lifetime of the struct. An `Rc` value isn't bound by any lifetimes and your change could cause lifetime issues, if the referenced config gets out of scope. I would suggest using an [Arc](https://doc.rust-lang.org/std/sync/struct.Arc.html)

Reply:
Thank you @Luro02 for reviewing. I'm still a little fuzzy on the problems this may change. I read more about [Rc](https://doc.rust-lang.org/1.30.0/book/2018-edition/ch15-04-rc.html). It enables multiple ownership and I could use an example where that is useful here. Our use case is simple, we create a config and pass a reference in to create any of the clients. In this case `APIClient` creates all the other clients, passing the reference.
``` rust
let apicli = crate::apis::client::APIClient::new(&config);
```

Reply:
```rust
fn create_client() -> Client {
    let config = Config::new();
    let client = Client::new(&config);
    client
} // config gets dropped here

fn main() {
    let client = create_client();
}
```
There are 2 solutions, either return the Config + Client or wrapping the config in a Arc/Rc. I think the idiomatic solution would be to let Client take ownership of the Config and then internally wrapping it in an Arc/Rc.